lib/spack/llnl/util/symlink.py

```diff
@@ -269,14 +269,14 @@ def _windows_create_hard_link(path: str, link: str):
         CreateHardLink(link, path)
 
 
-def readlink(path: str):
+def readlink(path: str, *, dir_fd=None):
```

Comment:
Change was made to achieve parity with actual `os.readlink`, previous implementation did not do this.